app/components/post_draft/draft_input/index.tsx

```diff
@@ -132,6 +150,102 @@ export default function DraftInput({
     const sendActionTestID = `${testID}.send_action`;
     const style = getStyleSheet(theme);
 
+    const persistenNotificationsEnabled = postPriority?.persistent_notifications && postPriority?.priority === PostPriorityType.URGENT;
+    const {noMentionsError, mentionsList} = useMemo(() => {
+        let error = false;
+        let mentions: string[] = [];
+        if (
+            channelType !== General.DM_CHANNEL &&
+            persistenNotificationsEnabled
+        ) {
+            mentions = (value.match(MENTIONS_REGEX) || []);
+            error = mentions.length === 0;
+        }
+
+        return {noMentionsError: error, mentionsList: mentions};
+    }, [channelType, persistenNotificationsEnabled, value]);
+
+    const handleSendMessage = useCallback(async () => {
+        if (persistenNotificationsEnabled) {
+            let title = '';
+            let description = '';
+            let error = true;
+            if (new RegExp(SPECIAL_MENTIONS_REGEX).test(value)) {
+                description = intl.formatMessage({
+                    id: 'persistent_notifications.error.special_mentions',
+                    defaultMessage: 'Cannot use @channel, @all or @here to mention recipients of persistent notifications.',
+                });
+            } else {
+                const formattedMentionsList = mentionsList.map((mention) => mention.slice(1));
+                const usersCount = database ? await getUsersCountFromMentions(database, formattedMentionsList) : 0;
```

Comment:
> but what happens if you paste the entire text (including mentions) instead of typing and one of them is not in the db.. tricky

@larkox, @enahum I don't think we need to trust the database we trust in the API instead. The API will return an error and won't submit the post if there are no mentions.
We do the same in the webapp. 
If you feel that we need to handle each and every case in the mobile can we do that in a separate ticket?

Reply:
Ok so we trust the API, great.. do the user get feedback telling them why the post failed to be created and do we handle the case not to retry it? Or am I missing something?

Reply:
I'd prefer to display the error message, what are your thoughts on that?

Reply:
@larkox Working with @koox00 on this.